nodecfg/node_config.go

```diff
@@ -102,19 +102,16 @@ func insertLogConfig(tx *sql.Tx, c *params.NodeConfig) error {
 }
 
 func insertNetworkConfig(tx *sql.Tx, c *params.NodeConfig) error {
-	if _, err := tx.Exec(`DELETE FROM network_config WHERE synthetic_id = 'id'`); err != nil {
-		return err
-	}
-
 	for _, network := range c.Networks {
 		_, err := tx.Exec(`
-	INSERT OR REPLACE INTO network_config (
-		chain_id, chain_name, rpc_url, block_explorer_url, icon_url, native_currency_name,
+        INSERT OR REPLACE INTO networks (
```

Comment:
I'm not sure, but this could affect older accounts migrating from when the node config was stored in settings to being stored in separate tables. To confirm this, we need to do the following:

1. Checkout desktop using release 0.3.0 (commit a8c37d98dacbf0997eb6e3124ea43dea240fd3c2), and rebuild status-go using that old version
2. Create a new account, login, logout
3. Update desktop to a PR that uses your status-go PR
4. Login succesfully with that old account
5. Logout succesfully with that old account
6. Login again succesfully with that old account

If there's an error there, then we need to modify these functions:
1. In [`nodeConfigUpgradeInserts`](https://github.com/status-im/status-go/blob/16864eb597bf54c9ce02bb2f294ecc5b59858bd9/nodecfg/node_config.go#L327-L346) ensure that we are calling an unmodified `insertNetworkConfig`. (the one that will be executed during the migration)
2. In [`nodeConfigNormalInserts`](https://github.com/status-im/status-go/blob/16864eb597bf54c9ce02bb2f294ecc5b59858bd9/nodecfg/node_config.go#L348-L372) call a modified `insertNetworkConfig` function that includes the `chain_color` and `short_name` columns 

Another option could be to just create an `insertNetworkChainColorAndShortName` that is called only in `nodeConfigNormalInserts`.

Reply:
Hey @richard-ramos what I did is below -
1. checked out to release/0.4.0 , created a new account logged in and out
2. Checked out to the desktop commit which uses this status-go PR. 
3. rm -rf vendor/ and make run -j8
4. while logging in, I get this error - 

Running: bin/nim_status_client
INFO [06-09|10:16:48.173] Status backend initialized               backend=geth version=0.100.1 commit=16864eb59
INF 2022-06-09 10:16:48.894+05:30 Version: 0.5.0                             topics="status-app" tid=82297 file=nim_status_client.nim:152
INF 2022-06-09 10:16:48.895+05:30 Commit: 519235522                          topics="status-app" tid=82297 file=nim_status_client.nim:153
INF 2022-06-09 10:16:48.895+05:30 Current date:                              topics="status-app" tid=82297 file=nim_status_client.nim:154 currentDateTime=2022-06-09T10:16:48+05:30
INF 2022-06-09 10:16:48.895+05:30 starting application controller...         topics="status-app" tid=82297 file=nim_status_client.nim:156
INF 2022-06-09 10:16:51.200+05:30 starting application...                    topics="status-app" tid=82297 file=nim_status_client.nim:164
ERROR[06-09|10:17:43.795] failed to initialize db                  package=status-go/api.GethStatusBackend err="table networks has no column named chain_color"
table networks has no column named chain_color
DBG 2022-06-09 10:17:43.796+05:30 Account logged in                          topics="accounts-service" tid=82297 file=service.nim:384
ERR 2022-06-09 10:17:43.924+05:30 node.crashed                               topics="signals-manager" tid=82297 file=signals_manager.nim:51 error="{\"type\":\"node.crashed\",\"event\":{\"error\":\"NoDiscovery is false, but ClusterConfig.BootNodes is empty\"}}"
ERR 2022-06-09 10:17:43.925+05:30 node.login                                 topics="signals-manager" tid=82297 file=signals_manager.nim:48 error="NoDiscovery is false, but ClusterConfig.BootNodes is empty"
ERR 2022-06-09 10:17:43.925+05:30 error:                                     topics="startup-controller" tid=82297 file=controller.nim:37 methodName=init errDesription="login error NoDiscovery is false, but ClusterConfig.BootNodes is empty"

Reply:
so the error isn't linked with the node config but the networks table?

Reply:
Hey @richard-ramos I created a new function insertNetworkConfigWithChainColorShortName and left insertNetworkConfig as it is and it works with the steps you described :)